test/python/sigopttest/parameters/from_json.py

```diff
@@ -91,14 +92,23 @@ def test_invalid_condition_values(self, parameter, invalid_parameter_json, condi
   @pytest.mark.parametrize(
     "invalid_parameter_json",
     [
-      dict(conditions=dict(x=None)),
       dict(conditions=dict(x=[])),
     ],
   )
   def test_condition_name_with_no_value(self, parameter, invalid_parameter_json, conditionals_map):
     with pytest.raises(BadParamError):
       set_parameter_conditions_from_json(parameter, invalid_parameter_json, conditionals_map)
```

Comment:
FWIW, I would expect these to eventually migrate to non-RequestError's as well, since this is just about JSON parsing. Fine to leave for subsequent work for now. And we can decide whether it makes sense to live here or in `libsigopt` - I would guess the latter.

Reply:
That's a great idea for the future, might start working on that as soon as tomorrow.